src/metabase/automagic_dashboards/core.clj

```diff
@@ -262,27 +237,36 @@
   [dimensions metrics order-by]
   (let [dimensions (set dimensions)]
     (for [[identifier ordering] (map first order-by)]
-      [(if (= ordering "ascending")
-         :asc
-         :desc)
-       (if (dimensions identifier)
-         [:dimension identifier]
-         [:aggregate-field (u/index-of #{identifier} metrics)])])))
+      [(if (dimensions identifier)
+          [:dimension identifier]
+          [:aggregation (u/index-of #{identifier} metrics)])
+        (if (= ordering "ascending")
+          :ascending
+          :descending)])))
+
+(defn merge-filters
```

Comment:
Would be good to reuse this instead https://github.com/metabase/metabase/blob/master/src/metabase/query_processor/middleware/expand_macros.clj#L115